Tool generates negative RunConfiguration.TestSessionTimeout which throws SettingsException

**Describe the bug**
After minor version update newest to the `dotnet-stryker` 1.5.1 tool, test execution fail. After enabling verbose logs, job fails because of negative `RunConfiguration.TestSessionTimeout` setting which throws an exception: `[Error] Microsoft.VisualStudio.TestPlatform.ObjectModel.SettingsException: Invalid settings 'RunConfiguration'.  Invalid value '-169088' specified for 'TestSessionTimeout'.` 

Example of failing [job](https://github.com/iamdmitrij/Tethos/runs/6066782447?check_suite_focus=true). 

Tool is invoked simply using `dotnet stryker` command:

```
dotnet stryker --config-file test/Tethos.Tests/stryker-config.json
  --version "$target_branch"
  --reporter "$extra_reporter"
  --reporter "cleartext"
  --reporter "html"
```
Full job definition can be found [here](https://github.com/iamdmitrij/Tethos/blob/c7380c597706fe8db854fa61be45ab6964031492/.github/workflows/stryker.yml#L70).

Note: Works fine with v1.5.0 with an [example job](https://github.com/iamdmitrij/Tethos/runs/6066941867?check_suite_focus=true).

**Logs**
```
[14:36:30 DBG] Runner 0: Using -169088 ms as test run timeout
[14:36:30 VRB] Runner 0: VsTest run settings set to: <RunSettings>
 <RunConfiguration>

  <MaxCpuCount>1</MaxCpuCount>
<TestSessionTimeout>-169088</TestSessionTimeout>

<DisableParallelization>true</DisableParallelization>
<DesignMode>false</DesignMode>
<BatchSize>1</BatchSize>

 </RunConfiguration>
<InProcDataCollectionRunSettings><InProcDataCollectors><InProcDataCollector friendlyName="StrykerCoverage" uri="https://stryker-mutator.io/" codebase="/home/runner/.dotnet/tools/.store/dotnet-stryker/1.5.1/dotnet-stryker/1.5.1/tools/net5.0/any/Stryker.DataCollector.dll" assemblyQualifiedName="Stryker.DataCollector.CoverageCollector, Stryker.DataCollector, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null">
<Configuration><Parameters><Mutant id='42' tests='23bae28c-17fa-9f1e-0f0b-0db4f21e11e3'/><Mutant id='48' tests='808036fc-5b2a-2989-183f-e2fe0e3d9ae3'/><Mutant id='49' tests='722834a5-e8af-ddf9-93d4-079787facdb7'/><Mutant id='50' tests='063e1e69-54f3-842e-9e24-a6f3f7cf1cf3'/><Mutant id='9' tests='ee944c96-c847-fee4-72b9-6b1df81aae40,fff51969-2e71-b134-9a61-f6a13c5b9d35'/><Mutant id='33' tests='e3676fae-eac6-ea07-5beb-305fae116731,8b444e74-5097-d7ee-035c-faba28a5203b'/><Mutant id='23' tests='50b8f8b5-6dde-937e-b591-2fdf4cca1200,27038c7c-aa26-336f-82e2-91e75aea6bcc,167df25e-fbb0-6e2c-e8ad-1d8c3a7aaad2'/><Mutant id='32' tests='a44ffca8-a476-2ecb-ebb9-a905695d6beb,0c1026b7-a8f0-687f-e114-e087c33962e3,247613aa-be8f-8138-83b9-9abbeeb2ffe7,53b1693d-09e4-c041-d8fb-89177d230449'/><Mutant id='29' tests='e8d95ee2-c242-fee3-9db7-a895cd0877c6,67130f64-00c4-167c-c3cf-a32507828d1f,8479cdc1-ea3a-5d0e-8f93-753004fbe602,6587978b-e611-3526-00ca-7c67c9d42934,e6a73c78-a210-fa14-fd48-77669ab794e7'/><Mutant id='17' tests='f6aeb162-41d0-ce35-7c84-fd15a4f7b932,5be922bd-b82c-ce41-e41c-2d95a9a88451,d0ec8f50-7ab9-dc5f-c054-583b924ac4bc,0babd1b3-4c43-cdac-0fca-f7fb2743d82e,6631d8b1-5f55-7051-a6df-5351b2b16530,2b18d6a1-e59e-a8d2-4d03-000177f21970'/><MutantControl  name='StrykerODz7lWODKfow9f2.MutantControl'/></Parameters></Configuration></InProcDataCollector></InProcDataCollectors></InProcDataCollectionRunSettings>
</RunSettings>
[14:36:30 WRN] Runner 0: VsTest did not report the end of test session in due time, it may have hang. Retrying
[14:36:30 INF] Time Elapsed 00:06:14.3630344
Unhandled exception. System.AggregateException: One or more errors occurred. (*** Failed to create a TestRunner, Stryker cannot recover from this!***)
 ---> 
*** Failed to create a TestRunner, Stryker cannot recover from this!***


   --- End of inner exception stack trace ---
   at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
Error: 30 VRB] Runner 0: [Error] Microsoft.VisualStudio.TestPlatform.ObjectModel.SettingsException: Invalid settings 'RunConfiguration'.  Invalid value '-169088' specified for 'TestSessionTimeout'.
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.RunConfiguration.FromXml(XmlReader reader)
   at System.Threading.Tasks.Parallel.ThrowSingleCancellationExceptionOrOtherException(ICollection exceptions, CancellationToken cancelToken, Exception otherException)
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.XmlRunSettingsUtilities.GetNodeValue[T](String settingsXml, String nodeName, Func`2 nodeParser)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.XmlRunSettingsUtilities.GetRunConfigurationNode(String settingsXml)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IList`1 list, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.TestPlatformHelpers.TestRequestManager.UpdateRunSettingsIfRequired(String runsettingsXml, IList`1 sources, IBaseTestEventsRegistrar registrar, String& updatedRunSettingsXml)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.TestPlatformHelpers.TestRequestManager.RunTests(TestRunRequestPayload testRunRequestPayload, ITestHostLauncher testHostLauncher, ITestRunEventsRegistrar testRunEventsRegistrar, ProtocolConfig protocolConfig)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body)
   at Microsoft.VisualStudio.TestPlatform.Client.DesignMode.DesignModeClient.<>c__DisplayClass23_0.<StartTestRun>b__0()
   at Stryker.Core.MutationTest.MutationTestProcess.TestMutants(IEnumerable`1 mutantsToTest)
   at Stryker.Core.MutationTest.MutationTestProcess.Test(IEnumerable`1 mutantsToTest)
   at Stryker.Core.StrykerRunner.RunMutationTest(IStrykerInputs inputs, ILoggerFactory loggerFactory, IProjectOrchestrator projectOrchestrator)
   at Stryker.CLI.StrykerCli.RunStryker(IStrykerInputs inputs) in /_/src/Stryker.CLI/Stryker.CLI/StrykerCLI.cs:line 93
   at Stryker.CLI.StrykerCli.<>c__DisplayClass10_0.<Run>b__0() in /_/src/Stryker.CLI/Stryker.CLI/StrykerCLI.cs:line 68
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.<>c__DisplayClass143_0.<OnExecute>b__0(CancellationToken _)
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.ExecuteAsync(String[] args, CancellationToken cancellationToken)
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
   at Stryker.CLI.StrykerCli.Run(String[] args) in /_/src/Stryker.CLI/Stryker.CLI/StrykerCLI.cs:line 74
   at Stryker.CLI.Program.Main(String[] args) in /_/src/Stryker.CLI/Stryker.CLI/Program.cs:line 14
Error: Process completed with exit code 134.
```

**Expected behavior**
`RunConfiguration.TestSessionTimeout` should not have negative value.

**Desktop:**
 - OS: Ubuntu
 -  .NET 6.0
 - Stryker Version 1.5.1

--- a/docs/technical-reference/testing-framework.md
+++ b/docs/technical-reference/testing-framework.md
@@ -39,31 +39,37 @@ between compile time theories (data is fixed at build time) and run time theorie
 Static discoveries are seen and processed as different test cases.
 
 #### Run time theories
-Run time theories are discovered as one test, disregarding the number of underlying data set.
-There is one test result per dataset, all associated with the same test.
+Run time theories may be discovered as one test, disregarding the number of underlying data set.
+During discovery phase, xUnit determines if it can generates a test case for each (theory) data set;
+if it can, those test cases are processed as usual; otherwise it will provide several results for the same test case.
+
+There is one test result per dataset, all associated with the same test case.
 
 Here is a summarized timeline of tests execution:
 ```
    xUnit runner calls data theory's data source to fetch each test case value(s)
 TestCase start event
    xUnit run test with first set of data
-   xUnit reports test results
-TestCase end event
    xUnit run test with second set of data
-   xUnit reports test reult
-   xUnit run test with third set of data
+   ...
+   xUnit run test with last set of data
+   xUnit reports first test results
+TestCase end event
+   xUnit reports second test reult
    ...
    xUnit reports last test result
 TestCase start event
    ...
 ```
+_Remark_: in this situation, each test result includes the TOTAL execution time (i.e. the time to test every data set) so one should not sum them.
+
 The difficulty here is that a lot happens between `testcase end` and `testcase start` events. 
 At coverage capture it is a problem, because there is a risk of spilling coverage information to the next tests: Stryker
 can only capture coverage information on `testcase end` events, and only in association with the current running test.
 So, every mutants covered between `testcase end` and `testcase start` will be associated with the **next** test.
 
-And during execution phase, it is impossible to predict when the test will really be over, so it is diffult to
-establish is the test was succesful.  
+And during execution phase, it is impossible to predict when the test will really be over, so it is difficult to
+establish if the test was succesful.  
 
 Also, if a mutation ends up changing a test case name - typically by changing the result of `ToTring()`, it will change the
 test identifier so this testcase will only run when running **all tests** and can no longer be executed in isolation, as

--- a/src/Stryker.Core/Stryker.Core.UnitTest/TestRunners/VsTestRunnersTest.cs
+++ b/src/Stryker.Core/Stryker.Core.UnitTest/TestRunners/VsTestRunnersTest.cs
@@ -48,7 +48,7 @@ namespace Stryker.Core.UnitTest.TestRunners
         private readonly TestProperty _coverageProperty;
         private readonly TestProperty _unexpectedCoverageProperty;
         private readonly Uri _xUnitUri;
-        private readonly TimeSpan _testDefaultDuration = TimeSpan.FromMilliseconds(100);
+        private readonly TimeSpan _testDefaultDuration = TimeSpan.FromSeconds(1);
 
         // initialize the test context and mock objects
         public VsTestRunnersTest()

--- a/src/Stryker.Core/Stryker.Core.UnitTest/TestRunners/VsTestRunnersTest.cs
+++ b/src/Stryker.Core/Stryker.Core.UnitTest/TestRunners/VsTestRunnersTest.cs
@@ -42,12 +42,13 @@ namespace Stryker.Core.UnitTest.TestRunners
         private readonly ProjectInfo _targetProject;
         private readonly MockFileSystem _fileSystem;
         private readonly Mutant _mutant;
-        private readonly List<TestCase> _testCases;
         private readonly Mutant _otherMutant;
         private readonly CsharpFolderComposite _projectContents;
         private readonly Uri _executorUri;
         private readonly TestProperty _coverageProperty;
         private readonly TestProperty _unexpectedCoverageProperty;
+        private readonly Uri _xUnitUri;
+        private readonly TimeSpan _testDefaultDuration = TimeSpan.FromMilliseconds(100);
 
         // initialize the test context and mock objects
         public VsTestRunnersTest()
@@ -75,6 +76,7 @@ namespace Stryker.Core.UnitTest.TestRunners
 </Project>";
             _testAssemblyPath = FilePathUtils.NormalizePathSeparators(Path.Combine(filesystemRoot, "_firstTest", "bin", "Debug", "TestApp.dll"));
             _executorUri = new Uri("exec://nunit");
+            _xUnitUri = new Uri("executor://xunit/VsTestRunner2/netcoreapp");
             var firstTest = BuildCase("T0");
             var secondTest = BuildCase("T1");
 
@@ -113,25 +115,24 @@ namespace Stryker.Core.UnitTest.TestRunners
                 ProjectContents = _projectContents
             };
 
-            _testCases = new List<TestCase> { firstTest, secondTest };
+            TestCases = new List<TestCase> { firstTest, secondTest };
         }
+
+        private List<TestCase> TestCases { get; }
 
         // simulate the discovery of tests
-        private static void DiscoverTests(ITestDiscoveryEventsHandler discoveryEventsHandler, ICollection<TestCase> tests, bool aborted)
-        {
+        private static void DiscoverTests(ITestDiscoveryEventsHandler discoveryEventsHandler, ICollection<TestCase> tests, bool aborted) =>
             Task.Run(() => discoveryEventsHandler.HandleDiscoveredTests(tests)).
                 ContinueWith((_, u) => discoveryEventsHandler.HandleDiscoveryComplete((int)u, null, aborted), tests.Count);
-        }
 
-        private TestCase BuildCase(string name) => new TestCase(name, _executorUri, _testAssemblyPath) { Id = new Guid() };
+        private TestCase BuildCase(string name) => new(name, _executorUri, _testAssemblyPath) { Id = new Guid() };
 
-        private TestCase FindOrBuildCase(string testResultId) => _testCases.FirstOrDefault(@t => t.FullyQualifiedName == testResultId) ?? BuildCase(testResultId);
+        private TestCase FindOrBuildCase(string testResultId) => TestCases.FirstOrDefault(t => t.FullyQualifiedName == testResultId) ?? BuildCase(testResultId);
 
         // mock a VsTest run. Provides test result one by one at 10 ms intervals
         // note: a lot of information is still missing (vs real VsTest). You will have to add them if your test requires them
         private void MoqTestRun(ITestRunEventsHandler testRunEvents, IReadOnlyList<TestResult> testResults,
-            TestCase timeOutTest = null)
-        {
+            TestCase timeOutTest = null) =>
             Task.Run(() =>
             {
                 var timer = new Stopwatch();
@@ -156,31 +157,32 @@ namespace Stryker.Core.UnitTest.TestRunners
                 testRunEvents.HandleTestRunComplete(
                     new TestRunCompleteEventArgs(new TestRunStatistics(testResults.Count, null), false, false, null,
                         null, timer.Elapsed),
-                    new TestRunChangedEventArgs(null, new TestResult[0], new List<TestCase>()),
+                    new TestRunChangedEventArgs(null, Array.Empty<TestResult>(), new List<TestCase>()),
                     null,
                     null);
             });
-        }
 
         // setup a simple (mock) test run where all tests fail or succeed
-        private void SetupMockTestRun(Mock<IVsTestConsoleWrapper> mockVsTest, bool testResult, EventWaitHandle synchroObject)
+        private void SetupMockTestRun(Mock<IVsTestConsoleWrapper> mockVsTest, bool testResult, EventWaitHandle synchroObject, ICollection<TestCase> testCases)
         {
-            var results = new List<(string, bool)>(_testCases.Count);
-            results.AddRange(_testCases.Select(t => (t.FullyQualifiedName, testResult)));
+            var results = new List<(string, bool)>(testCases.Count);
+            results.AddRange(testCases.Select(t => (t.FullyQualifiedName, testResult)));
             SetupMockTestRun(mockVsTest, results, synchroObject);
         }
 
         private void SetupMockTestRun(Mock<IVsTestConsoleWrapper> mockVsTest, IEnumerable<(string id, bool success)> testResults,
-            EventWaitHandle synchroObject)
+            EventWaitHandle synchroObject, ICollection<TestCase> testCases = null)
         {
             var results = new List<TestResult>();
+            testCases ??= TestCases;
             foreach (var (testResultId, success) in testResults)
             {
-                var testCase = FindOrBuildCase(testResultId);
+                var testCase = testCases.FirstOrDefault(t => t.FullyQualifiedName == testResultId) ?? BuildCase(testResultId);
                 results.Add(new TestResult(testCase)
                 {
                     Outcome = success ? TestOutcome.Passed : TestOutcome.Failed,
-                    ComputerName = "."
+                    ComputerName = ".",
+                    Duration = _testDefaultDuration
                 });
             }
             mockVsTest.Setup(x =>
@@ -344,8 +346,9 @@ namespace Stryker.Core.UnitTest.TestRunners
                     endProcess.Set();
                 }).Returns(Task.CompletedTask);
 
-        private Mock<IVsTestConsoleWrapper> BuildVsTestRunner(StrykerOptions options, WaitHandle endProcess, out VsTestRunner runner, bool suceed = true)
+        private Mock<IVsTestConsoleWrapper> BuildVsTestRunner(StrykerOptions options, WaitHandle endProcess, out VsTestRunner runner, ICollection<TestCase> testCases=null, bool suceed = true)
         {
+            testCases ??= TestCases;
             var mockedVsTestConsole = new Mock<IVsTestConsoleWrapper>(MockBehavior.Strict);
             mockedVsTestConsole.Setup(x => x.StartSession());
             mockedVsTestConsole.Setup(x => x.InitializeExtensions(It.IsAny<List<string>>()));
@@ -355,7 +358,7 @@ namespace Stryker.Core.UnitTest.TestRunners
                     It.IsAny<string>(),
                     It.IsAny<ITestDiscoveryEventsHandler>())).Callback(
                 (IEnumerable<string> _, string _, ITestDiscoveryEventsHandler discoveryEventsHandler) =>
-                    DiscoverTests(discoveryEventsHandler, _testCases, false));
+                    DiscoverTests(discoveryEventsHandler, testCases, false));
 
             runner = new VsTestRunner(
                 options,
@@ -376,7 +379,7 @@ namespace Stryker.Core.UnitTest.TestRunners
             using var endProcess = new EventWaitHandle(true, EventResetMode.ManualReset);
             BuildVsTestRunner(new StrykerOptions(), endProcess, out var runner);
             // make sure we have discovered first and second tests
-            runner.DiscoverTests().Count.ShouldBe(_testCases.Count);
+            runner.DiscoverTests().Count.ShouldBe(TestCases.Count);
         }
 
         [Fact]
@@ -395,7 +398,7 @@ namespace Stryker.Core.UnitTest.TestRunners
         public void HandleVsTestCreationFailure()
         {
             using var endProcess = new EventWaitHandle(false, EventResetMode.ManualReset);
-            var mockVsTest = BuildVsTestRunner(new StrykerOptions(), endProcess, out var runner, false);
+            var mockVsTest = BuildVsTestRunner(new StrykerOptions(), endProcess, out var runner, TestCases, false);
             SetupMockTestRun(mockVsTest, new[] { ("T0", false), ("T1", true) }, endProcess);
 
             Action action = () => runner.InitialTest();
@@ -407,7 +410,7 @@ namespace Stryker.Core.UnitTest.TestRunners
         {
             using var endProcess = new EventWaitHandle(false, EventResetMode.ManualReset);
             var mockVsTest = BuildVsTestRunner(new StrykerOptions(), endProcess, out var runner);
-            SetupMockTestRun(mockVsTest, true, endProcess);
+            SetupMockTestRun(mockVsTest, true, endProcess, TestCases);
             var result = runner.TestMultipleMutants(null, new []{_mutant}, null);
             // tests are successful => run should be successful
             result.FailingTests.IsEmpty.ShouldBeTrue();
@@ -420,7 +423,7 @@ namespace Stryker.Core.UnitTest.TestRunners
             using (var endProcess = new EventWaitHandle(false, EventResetMode.AutoReset))
             {
                 var mockVsTest = BuildVsTestRunner(options, endProcess, out var runner);
-                SetupMockTestRun(mockVsTest, false, endProcess);
+                SetupMockTestRun(mockVsTest, false, endProcess, TestCases);
                 var result = runner.TestMultipleMutants(null, new []{_mutant}, null);
                 // run is failed
                 result.FailingTests.IsEmpty.ShouldBeFalse();
@@ -460,7 +463,7 @@ namespace Stryker.Core.UnitTest.TestRunners
                 var mockVsTest = BuildVsTestRunner(options, endProcess, out var runner);
 
                 mockVsTest.Setup(x => x.CancelTestRun()).Verifiable();
-                SetupMockTestRun(mockVsTest, false, endProcess);
+                SetupMockTestRun(mockVsTest, false, endProcess, TestCases);
 
                 var result = runner.TestMultipleMutants(null, new []{_mutant},  ((_, _, _, _) => false));
                 // verify Abort has been called
@@ -487,7 +490,7 @@ namespace Stryker.Core.UnitTest.TestRunners
                 runner.CaptureCoverage(_targetProject.ProjectContents.Mutants);
                 _mutant.CoveringTests.IsEmpty.ShouldBe(false);
                 var id = _mutant.CoveringTests.GetGuids().First();
-                _testCases.Find(t => t.Id == id)?.DisplayName.ShouldBe("T0");
+                TestCases.Find(t => t.Id == id)?.DisplayName.ShouldBe("T0");
             }
         }
 
@@ -510,8 +513,8 @@ namespace Stryker.Core.UnitTest.TestRunners
                 // one mutant is covered by tests 0 and 1
                 _mutant.CoveringTests.IsEmpty.ShouldBe(false);
 
-                _testCases.Find(t => t.Id == testIds[0])?.DisplayName.ShouldBe("T0");
-                _testCases.Find(t => t.Id == testIds[1])?.DisplayName.ShouldBe("T1");
+                TestCases.Find(t => t.Id == testIds[0])?.DisplayName.ShouldBe("T0");
+                TestCases.Find(t => t.Id == testIds[1])?.DisplayName.ShouldBe("T1");
 
                 // verify Abort has been called
                 result.FailingTests.IsEmpty.ShouldBeTrue();
@@ -582,7 +585,7 @@ namespace Stryker.Core.UnitTest.TestRunners
                 _otherMutant.ResetCoverage();
                 runner.CaptureCoverage(_targetProject.ProjectContents.Mutants);
 
-                SetupMockTestRun(mockVsTest, false, endProcess);
+                SetupMockTestRun(mockVsTest, false, endProcess, TestCases);
                 // mutant 0 is covered
                 _mutant.IsStaticValue.ShouldBeTrue();
                 var result = runner.TestMultipleMutants(null, new []{_mutant}, null);
@@ -612,10 +615,10 @@ namespace Stryker.Core.UnitTest.TestRunners
                 var mockVsTest = BuildVsTestRunner(options, endProcess, out var runner);
                 // make sure we have 4 mutants
                 _projectContents.Add(new CsharpFileLeaf { Mutants = new[] { new Mutant { Id = 2 }, new Mutant { Id = 3 } } });
-                _testCases.Add(new TestCase("T2", _executorUri, _testAssemblyPath));
-                _testCases.Add(new TestCase("T3", _executorUri, _testAssemblyPath));
+                TestCases.Add(new TestCase("T2", _executorUri, _testAssemblyPath));
+                TestCases.Add(new TestCase("T3", _executorUri, _testAssemblyPath));
 
-                var ranTests = new TestsGuidList(_testCases.Select(t => t.Id));
+                var ranTests = new TestsGuidList(TestCases.Select(t => t.Id));
                 var testRunResult = new TestRunResult(ranTests, TestsGuidList.NoTest(), TestsGuidList.NoTest(),
                     string.Empty, TimeSpan.Zero);
                 var input = new MutationTestInput
@@ -684,6 +687,24 @@ namespace Stryker.Core.UnitTest.TestRunners
         }
 
         [Fact]
+        public void HandleMultipleTestResultsForXUnit()
+        {
+            var options = new StrykerOptions();
+            using var endProcess = new EventWaitHandle(false, EventResetMode.ManualReset);
+            var tests = new List<TestCase>
+            {
+                new("X0", _xUnitUri, "test.cs"),
+                new("X1", _xUnitUri, "test.cs")
+            };
+            var mockVsTest = BuildVsTestRunner(options, endProcess, out var runner, tests);
+            // assume 2 results for T0
+            SetupMockTestRun(mockVsTest, new[] { ("X0", true), ("X1", true), ("X0", true) , ("X0", true), ("X0", true)}, endProcess, tests);
+            var result = runner.InitialTest();
+            // the duration should be less than 3 times the (test) default duration
+            result.Duration.ShouldBeLessThan(_testDefaultDuration.Duration()*3);
+        }
+
+        [Fact]
         public void HandleFailureWithinMultipleTestResults()
         {
             var options = new StrykerOptions();
@@ -700,13 +721,13 @@ namespace Stryker.Core.UnitTest.TestRunners
             result = runner.TestMultipleMutants(null, new []{_mutant}, null);
             result.RanTests.IsEveryTest.ShouldBeTrue();
             result.FailingTests.IsEmpty.ShouldBeFalse();
-            result.FailingTests.GetGuids().ShouldContain(_testCases[0].Id);
+            result.FailingTests.GetGuids().ShouldContain(TestCases[0].Id);
             // test session will fail on the other test result
             SetupMockTestRun(mockVsTest, new[] { ("T0", true), ("T0", false), ("T1", true) }, endProcess);
             result = runner.TestMultipleMutants(null, new []{_mutant}, null);
             result.RanTests.IsEveryTest.ShouldBeTrue();
             result.FailingTests.IsEmpty.ShouldBeFalse();
-            result.FailingTests.GetGuids().ShouldContain(_testCases[0].Id);
+            result.FailingTests.GetGuids().ShouldContain(TestCases[0].Id);
         }
 
         [Fact]
@@ -726,7 +747,7 @@ namespace Stryker.Core.UnitTest.TestRunners
 
             result.FailingTests.IsEmpty.ShouldBeTrue();
             result.TimedOutTests.Count.ShouldBe(1);
-            result.TimedOutTests.GetGuids().ShouldContain(_testCases[0].Id);
+            result.TimedOutTests.GetGuids().ShouldContain(TestCases[0].Id);
             result.RanTests.IsEveryTest.ShouldBeFalse();
         }
 
@@ -746,7 +767,7 @@ namespace Stryker.Core.UnitTest.TestRunners
             result = runner.TestMultipleMutants(null, new []{_mutant}, null);
             result.RanTests.IsEveryTest.ShouldBeTrue();
             result.FailingTests.IsEmpty.ShouldBeFalse();
-            result.FailingTests.GetGuids().ShouldContain(_testCases[0].Id);
+            result.FailingTests.GetGuids().ShouldContain(TestCases[0].Id);
         }
 
         [Fact]

--- a/src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestDescription.cs
+++ b/src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestDescription.cs
@@ -10,7 +10,7 @@ namespace Stryker.Core.TestRunners.VsTest
     public class VsTestDescription
     {
         private readonly ICollection<TestResult> _initialResults = new List<TestResult>();
-        private int _subCases = 0;
+        private int _subCases;
 
         public VsTestDescription(TestCase testCase)
         {
@@ -26,18 +26,25 @@ namespace Stryker.Core.TestRunners.VsTest
                 {
                     return TestFramework.nUnit;
                 }
-                if (Case.Properties.Any(p => p.Id == "XunitTestCase"))
-                {
-                    return TestFramework.xUnit;
-                }
-
-                return TestFramework.msTest;
+                return Case.ExecutorUri.AbsoluteUri.Contains("xunit") ? TestFramework.xUnit : TestFramework.msTest;
             }
         }
 
         public TestDescription Description { get; }
 
-        public TimeSpan InitialRunTime => _initialResults.Aggregate(new TimeSpan(), (current, result) => current + result.Duration);
+        public TimeSpan InitialRunTime
+        {
+            get
+            {
+                if (Framework == TestFramework.xUnit)
+                {
+                    // xUnit returns the run time for the case within each result, so the first one is sufficient
+                    return _initialResults.First().Duration;
+                }
+
+                return TimeSpan.FromTicks(_initialResults.Sum(t => t.Duration.Ticks));
+            }
+        }
 
         public Guid Id => Case.Id;
 
@@ -45,15 +52,9 @@ namespace Stryker.Core.TestRunners.VsTest
 
         public int NbSubCases => Math.Max(_subCases, _initialResults.Count);
 
-        public void RegisterInitialTestResult(TestResult result)
-        {
-            _initialResults.Add(result);
-        }
+        public void RegisterInitialTestResult(TestResult result) => _initialResults.Add(result);
 
-        public void AddSubCase()
-        {
-            _subCases++;
-        }
+        public void AddSubCase() => _subCases++;
 
         protected bool Equals(VsTestDescription other) => Equals(Case.Id, other.Case.Id);
 

--- a/src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs
+++ b/src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs
@@ -91,8 +91,10 @@ namespace Stryker.Core.TestRunners.VsTest
                     _logger.LogWarning("{RunnerId}: Initial test run encounter a unexpected test case ({TestCaseDisplayName}), mutation tests may be inaccurate. Disable coverage analysis if your have doubts.",
                         RunnerId, result.TestCase.DisplayName);
                 }
+
                 _vsTests[result.TestCase.Id].RegisterInitialTestResult(result);
             }
+
             // get the test results, but prevent compression of 'all tests'
             return BuildTestRunResult(testResults, int.MaxValue, false);
         }
@@ -137,8 +139,7 @@ namespace Stryker.Core.TestRunners.VsTest
                     if (timeoutCalc != null && testCases != null)
                     {
                         // compute time out
-                        var duration = (int)testCases.Select(id => _vsTests[id].InitialRunTime.TotalMilliseconds).Sum();
-                        timeOutMs = timeoutCalc.CalculateTimeoutValue(duration);
+                        timeOutMs = timeoutCalc.CalculateTimeoutValue((int)testCases.Sum(id => _vsTests[id].InitialRunTime.TotalMilliseconds));
                     }
                 }
                 else
@@ -208,7 +209,7 @@ namespace Stryker.Core.TestRunners.VsTest
                 _logger.LogTrace($"{RunnerId}: Initial Test session reports 0 result and 0 stuck tests.");
             }
 
-            var duration = testResults.TestResults.Aggregate(new TimeSpan(), (span, result) => span.Add(result.Duration));
+            var duration =  TimeSpan.FromTicks(_vsTests.Values.Sum(t => t.InitialRunTime.Ticks));
 
             var message = string.Join(Environment.NewLine,
                 resultAsArray.Where(tr => !string.IsNullOrWhiteSpace(tr.ErrorMessage))

